test/routines_test.rb

```diff
@@ -39,3 +39,45 @@ def initialize
     end
   end
 end
+
+describe "#ValidNicName" do
+  subject(:routines) { RoutinesTestClass.new }
+
+  it "succeedes for valid names" do
+    expect(routines.ValidNicName("eth0")).to be true
+    expect(routines.ValidNicName("eth_0")).to be true
+    expect(routines.ValidNicName("eth-0")).to be true
+    expect(routines.ValidNicName("eth.0")).to be true
+    expect(routines.ValidNicName("eth:0")).to be true
+  end
+
+  it "fails in case of long name" do
+    expect(routines.ValidNicName("0123456789012345")). to be false
+  end
+
+  it "fails when it contains invalid character" do
+    expect(routines.ValidNicName("eth0?")).to be false
+  end
+end
+
+describe "#DeviceName" do
+  subject(:routines) { RoutinesTestClass.new }
+  let(:hwinfo) do
+    {
+      "sub_vendor" => "hw_vendor",
+      "sub_device" => "hw_device"
+    }
+  end
+
+  it "returns empty string when nothing is defined" do
+    expect(routines.DeviceName({})).to be_empty
+  end
+
+  it "returns description containing model name when known" do
+    expect(routines.DeviceName(hwinfo.merge("model" => "hw_model"))).to eql "hw_model"
+  end
+
+  it "returns description build from vendor and device details when model name is uknown" do
+    expect(routines.DeviceName(hwinfo)).to eql "hw_vendor hw_device"
+  end
```

Comment:
I've somehow expected comments like this. But I had to try to pass with only converting old testsuites ;-) ok, ok will extend it too ;-)